www/public/repl/repl.js

```diff
@@ -165,11 +165,48 @@ async function processInputQueue() {
 // Callbacks to JS from Rust
 // ----------------------------------------------------------------------------
 
+var ROC_PANIC_INFO = null;
+
+function send_panic_msg_to_js(rocstr_ptr, panic_tag) {
+    const { memory } = repl.app.exports;
+
+    const rocStrBytes = new Int8Array(memory.buffer, rocstr_ptr, 12);
+    const finalByte = rocStrBytes[11]
+
+    let stringBytes = "";
+    if (finalByte < 0) {
+        // small string
+        const length = finalByte ^ 0b1000_0000;
+        stringBytes = new Uint8Array(memory.buffer, rocstr_ptr, length);
+    } else {
+        // big string
+        const rocStrWords = new Uint32Array(memory.buffer, rocstr_ptr, 3);
+        const [ptr, len, _cap] = rocStrWords;
+
+        const SEAMLESS_SLICE_BIT = 1 << 31;
+        const length = len & (~SEAMLESS_SLICE_BIT);
```

Comment:
This bitwise op should work fine. It can't be a negative number because we read it out of an unsigned array. 👍